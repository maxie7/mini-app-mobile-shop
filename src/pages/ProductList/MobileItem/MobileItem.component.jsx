import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { MobileItemContainer } from "./MobileItem.styles";
import { Image, Frame } from "arwes";
import { NavLink } from "react-router-dom";

const MobileItem = ({ id, brand, model, price, imgUrl }) => {
  const [showFrame, setShowFrame] = useState(false);

  useEffect(() => {
    const timer = setTimeout(() => {
      setShowFrame(true);
    }, 600);
    return () => clearTimeout(timer);
  }, []);

  return (
    <Frame
      show={showFrame}
      animate={true}
      level={3}
      corners={4}
      layer='primary'
    >
      <MobileItemContainer key={ id }>
        <NavLink to={'/product/' + id}>
          <Image animate resources={ imgUrl }>
            <p>{ brand }</p>
            <p>{ model }</p>
            { price && <p>{ price } â‚¬</p> }
          </Image>
        </NavLink>
      </MobileItemContainer>
    </Frame>
  )
}

MobileItem.propTypes = {
  id:     PropTypes.string,
  brand:  PropTypes.string,
  model:  PropTypes.string,
  price:  PropTypes.string,
  imgUrl: PropTypes.string
};

export default MobileItem;
