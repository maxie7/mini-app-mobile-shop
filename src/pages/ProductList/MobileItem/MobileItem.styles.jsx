import styled from 'styled-components';

export const MobileItemContainer = styled.div`
  background: #021114;
  display: flex;
  flex-direction: column;
  font-family: "Electrolize", sans-serif;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  padding: 20px 40px;
  transform: translateZ(0);
  transition: transform 0.25s ease-out;
  
  &:hover {
    transform: scale(1.1);
  }
  
  @media screen and (max-width: 800px) {
    width: 42vw;
    
    &:hover {
      transform: unset;
  }
`;
