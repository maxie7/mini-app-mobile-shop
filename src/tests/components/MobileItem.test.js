import React from "react";
import {shallow} from "enzyme";
import MobileItem from "../../pages/ProductList/MobileItem/MobileItem.component";

it('should render MobileItem section', () => {
  expect(shallow(<MobileItem />)).toMatchSnapshot();
});