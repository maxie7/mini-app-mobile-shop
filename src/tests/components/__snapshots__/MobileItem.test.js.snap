// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MobileItem section 1`] = `ShallowWrapper {}`;
